tests/functional/aws-node-sdk/test/versioning/replicationBucket.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+const async = require('async');
+
+const withV4 = require('../support/withV4');
+const BucketUtility = require('../../lib/utility/bucket-util');
+
+const bucketName = `versioning-bucket-${Date.now()}`;
+
+
+function checkError(err, code) {
+    assert.notEqual(err, null, 'Expected failure but got success');
+    assert.strictEqual(err.code, code);
+}
+
+function checkNoError(err) {
+    assert.ifError(err, `Expected success, got error ${JSON.stringify(err)}`);
+}
+
+function testVersioning(s3, versioningStatus, replicationStatus, cb) {
+    const versioningParams = { Bucket: bucketName,
+        VersioningConfiguration: { Status: versioningStatus } };
+    const replicationParams = {
+        Bucket: bucketName,
+        ReplicationConfiguration: {
+            Role: 'arn:aws:iam::123456789012:role/examplerole,' +
+            'arn:aws:iam::123456789012:role/examplerole',
+            Rules: [
+                {
+                    Destination: {
+                        Bucket: 'arn:aws:s3:::destinationbucket',
+                        StorageClass: 'STANDARD',
+                    },
+                    Prefix: '',
+                    Status: replicationStatus,
+                },
+            ],
+        },
+    };
+    async.waterfall([
+        cb => s3.putBucketReplication(replicationParams, e => cb(e)),
+        cb => s3.putBucketVersioning(versioningParams, e => cb(e)),
```

Comment:
Might not need the extra wrapper method on the callbacks, sometimes you do with dealing with `this`...

```suggestion
        cb => s3.putBucketReplication(replicationParams, cb),
        cb => s3.putBucketVersioning(versioningParams, cb),
```

Reply:
Ah this, it's needed as this is a waterfall and I would have to feed both result and response to the next step